sw/device/lib/crypto/drivers/hmac.c

```diff
@@ -256,6 +266,36 @@ static void msg_fifo_write(const uint8_t *message, size_t message_len) {
   }
 }
 
+/**
+ * Temporary workaround linked to issue #24767
+ *
+ * The HMAC HWIP is not tell to stop. This will cause the HW to be in a non
+ * expected state, which could be exited by trigerring a simple HASH process
+ * operation. Before doing that, the context should be saved (message length
+ * and digest). This context is only available after a duration equivalent to
+ * 64 clock cycles in SHA2-256 and 80 clock cycles in SHA2-384/512, after the
+ * message length is on a block boundary (512 for SHA2-256 or 1024 bits for
+ * SHA2-384/512).
+ *
+ * @param[out] ctx Context to which values are written.
+ */
+static void tmp_avoid_hw_hang(hmac_ctx_t *ctx) {
+  // Insert delay which should be equivalent to at least 80 clock cycles
+  for (size_t i = 0; i < 100; i = launder32(i + 1)) {
+    // NULL statement
+    ;
+  }
```

Comment:
I think you can use the `IBEX_SPIN_FOR` macro here to ensure the compiler doesn't "helpfully" optimize away the loop:
```suggestion
uint64_t delay_cycles = 80; // or maybe 100 or so to be safe
uint64_t delay_usec = udiv64_slow(delay_cycles * 1000000, kClockFreqCpuHz, NULL);
IBEX_SPIN_FOR(delay_usec);
```

Reply:
Also, can we use the constant defined above for this, i.e., `kNumIterTimeout`?

Reply:
Thanks Jade for the trick, that's much better. And yes Pirmin, it makes sense to use this constant as it refers to the same HW constraint.

Reply:
@jadephilipoom I have implemented something else based on what we've discussed together offline. Can you please have a look at these changes?

Reply:
this constant can be lower, as loop is 2 instructions, so 80 cycles needs only 40 iterations max and you already did something else after sending to FIFO (like return from function, storing/restoring regs), etc.